sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala

```diff
@@ -47,10 +47,13 @@ private[connector] trait TestV2SessionCatalogBase[T <: Table] extends Delegating
       tables.get(ident)
     } else {
       // Table was created through the built-in catalog
-      val t = super.loadTable(ident)
-      val table = newTable(t.name(), t.schema(), t.partitioning(), t.properties())
-      tables.put(ident, table)
-      table
+      super.loadTable(ident) match {
+        case v1Table: V1Table if v1Table.v1Table.tableType == CatalogTableType.VIEW => v1Table
```

Comment:
@imback82 . This is orthogonal to `Fix an invalid test in DataSourceV2SQLSuite`. Could you make a PR for this one independently?

Reply:
or we should at least mention it in the PR description, to explain why it's needed to fix the test.

Reply:
Updated the PR description.